Contest_Archive_Format.md

```diff
@@ -16,708 +17,151 @@ disk, including:
+  `contest.json`. (Notice the singular form).
+- The JSON returned from the endpoint `/contests/<id>/<endpoint>` is stored as
+  `<endpoint>.json`.
+- The NDJSON returned from the endpoint `/contests/<id>/<endpoint>` is stored as
+  `<endpoint>.ndjson`. (The only such endpoint is `event-feed`) 
+
+
+Files referenced to in `contest.json` are stored as `contest/<filename>`, and
+files referenced to in `<endpoint>.json` as `<endpoint>/<id>/<filename>`,
+where:
+- `<id>` is the ID of the endpoint object the reference is in.
+- `<filename>` is the filename specified in the file reference object.
+
+Note that the API specification requires that filenames are unique within
+endpoint objects, so this is always possible.
+
+The href in the file reference may not be valid and should be ignored, but it
+does not have to be removed.
+
+Optionally one could create a Shallow Archive by not storing the files, but
+then the hrefs must be valid. This could be useful in some cases where the
+size of the archive matters.
+
+In some cases it could make sense to merge multiple API sources (of the same
+contest) in a single archive. One example of this would be a contest that was
+running with a primary and [shadow](ccs_system_requirements#shadow-mode) CCS.
+Typically in such cases, most of the data will be identical (or at least the
+differences irrelevant), so only the data that differs (in relevant ways)
+needs to be stored. If there is an additonal source for `<endpoint>` it is
+stored as above, but wherever `<endpoint>` is used in the path, instead use
+`<endpoint>_<system>` where `<system>` is a unique name for the additional
+source.
+
+## Exampole uses
+
+The rest of the document describes use cases for the Archive Format, listing
+which data would be required for that use case.
+
+### CCS configuration
+
+Used for configuring a CCS (primary or shadow) before a contest. Suitable for
+download from a registration system or similar tool.
+
+Required endpoints:
+- contests
+- languages
+- problems
+- teams
+- accounts
+
+Optional endpoints:
+- judgement-types
+- groups
+- organizations
+- people
+
+#### Example file listing
+
+```
+contest.json
+contest/banner.png
+contest/logo.png
+judgement-types.json
+languages.json
+problems.json
+problems/problemA/problemA.zip
+problems/problemA/problemA.pdf
+problems/problemB/problemB.zip
+problems/problemB/problemB.pdf
+...
+groups.json
+organizations.json
+organizations/kth.se/logo56x56.png
+organizations/kth.se/logo160x160.png
+organizations/baylor.edu/logo56x56.png
+organizations/baylor.edu/logo160x160.png
+...
+teams.json
+teams/team-001/photo.jpg
+teams/team-002/photo.jpg
+teams/team-003/photo.jpg
+...
+people.json
+people/john-smith/photo.jpg
+people/jane-doe/photo.jpg
+....
+accounts.json
+```
+
+### Registration upload
+
+Used for uploading local registration data to a central registration system
+(such as the ICPC CMS).
+
+Required endpoints:
+- teams
+- people
+
```

Comment:
Does this not require (or optional) org info?

Reply:
Yes... it annoyingly does. Thanks.

My thinking was that the upload should not support adding new universities to the ICPC database (to keep things simple), but since the only university reference in `teams.json` is a `organization_id` (which could just be an arbitrary string) you do need `organizations.json`, if only to map `organization_id` to `icpc_id`.

I have added this.